system/jlib/jtrace.cpp

```diff
@@ -107,6 +108,260 @@ class NoopSpanExporterFactory
     }
 };
 
+/**
+ * Converts an OpenTelemetry span kind to its string representation.
+ * These are OTel span kinds defined in api/include/opentelemetry/trace/span.h,
+ * not HPCC JLib CSpan kinds
+ *
+ * @param spanKind The OpenTelemetry span kind to convert.
+ * @return The string representation of the OpenTelemetry span kind.
+ */
+const char * spanKindToString(opentelemetry::trace::SpanKind spanKind)
+{
+    switch (spanKind)
+    {
+        case opentelemetry::trace::SpanKind::kClient:
+            return "Client";
+        case opentelemetry::trace::SpanKind::kServer:
+            return "Server";
+        case opentelemetry::trace::SpanKind::kProducer:
+            return "Producer";
+        case opentelemetry::trace::SpanKind::kConsumer:
+            return "Consumer";
+        case opentelemetry::trace::SpanKind::kInternal:
+            return "Internal";
+        default:
+            return "Unknown";
+    }
+}
+
+class JLogSpanExporter final : public opentelemetry::sdk::trace::SpanExporter
+{
+private:
+    SpanLogFlags logFlags = SpanLogFlags::LogNone;
+
+public:
+    JLogSpanExporter(SpanLogFlags spanLogFlags) : shutDown(false)
+    {
+        logFlags = spanLogFlags;
+    }
+
+    /**
+    * @return Returns a unique pointer to an empty recordable object
+    */
+    virtual std::unique_ptr<opentelemetry::sdk::trace::Recordable> MakeRecordable() noexcept override
+    {
+        return std::unique_ptr<opentelemetry::sdk::trace::Recordable>(new opentelemetry::sdk::trace::SpanData());
+    }
+
+    /**
+    * Export - Formats recordable spans in HPCC Jlog format and reports to JLog
+    *
+    * @param recordables
+    * @return Always returns success
+    */
+    opentelemetry::sdk::common::ExportResult Export(
+      const nostd::span<std::unique_ptr<opentelemetry::sdk::trace::Recordable>> &recordables) noexcept override
+    {
+        if (isShutDown())
+        {
+            ERRLOG("JLog Trace Exporter: Export failed, exporter is shutdown");
```

Comment:
Why do you think the logging useful?  

The reason it is unhelpful is that it would only be called if there is an internal problem in the open telemetry libraries.  Either a race condition or a problem in the logic.

It could be happening in closedown - which would potentially cause the system to core if the logging manager has been destroyed.  (We have had cores for this reason.)